pkg/capture/capture.go

```diff
@@ -43,172 +33,80 @@ var defaultSourceInitFn = func(c *Capture) (capture.Source, error) {
 	)
 }
 
-//////////////////////// Ancillary capturetypes ////////////////////////
-
-// ErrorMap stores all encountered pcap errors and their number of occurrence
-type ErrorMap map[string]int
-
-// String prints the errors that occurred during capturing
-func (e ErrorMap) String() string {
-	var errs = make([]string, len(e))
+// sourceInitFn denotes the function used to initialize a capture source,
+// providing the ability to override the default behavior, e.g. in mock tests
+type sourceInitFn func(*Capture) (capture.Source, error)
 
-	i := 0
-	for err, count := range e {
-		errs[i] = fmt.Sprintf("%s (%d)", err, count)
-		i++
-	}
-	sort.Slice(errs, func(i, j int) bool {
-		return errs[i] < errs[j]
-	})
-	return strings.Join(errs, "; ")
+// Captures denotes a named set of Capture instances, wrapping a map and the
+// required synchronization of all its actions
+type Captures struct {
```

Comment:
Created new type that wraps the cumbersome map[string]*Capture and provides convenience methods to handle all the locking without the caller having to worry about it.

Reply:
Does it need to be public?

Reply:
Fair point - no, it doesn't. Will make it private then!